src/query/storages/system/src/temp_files_table.rs

```diff
@@ -151,6 +141,165 @@ impl TempFilesTable {
             ..Default::default()
         };
 
-        AsyncOneBlockSystemTable::create(Self { table_info })
+        Arc::new(Self { table_info })
+    }
+}
+
+struct TempFilesTableAsyncSource {
+    finished: bool,
+    context: Arc<dyn TableContext>,
+    operator: Operator,
+    location_prefix: String,
+    entries_processed: usize,
+    limit: usize,
+    lister: Option<Lister>,
+}
+
+impl TempFilesTableAsyncSource {
+    pub fn create(
+        ctx: Arc<dyn TableContext>,
+        output: Arc<OutputPort>,
+        push_downs: Option<PushDownInfo>,
+    ) -> Result<ProcessorPtr> {
+        let tenant = ctx.get_tenant();
+        let location_prefix = format!("{}/", query_spill_prefix(tenant.tenant_name(), ""));
+        let limit = push_downs
+            .as_ref()
+            .and_then(|x| x.limit)
+            .unwrap_or(usize::MAX);
+
+        AsyncSourcer::create(ctx.clone(), output, TempFilesTableAsyncSource {
+            finished: false,
+            context: ctx,
+            operator: DataOperator::instance().operator(),
+            location_prefix,
+            entries_processed: 0,
+            limit,
+            lister: None,
+        })
+    }
+
+    fn build_block(
+        names: Vec<String>,
+        file_lens: Vec<u64>,
+        file_last_modifieds: Vec<Option<i64>>,
+    ) -> DataBlock {
+        let row_number = names.len();
+        DataBlock::new(
+            vec![
+                BlockEntry::new(
+                    DataType::String,
+                    Value::Scalar(Scalar::String("Spill".to_string())),
+                ),
+                BlockEntry::new(
+                    DataType::String,
+                    Value::Column(StringType::from_data(names)),
+                ),
+                BlockEntry::new(
+                    DataType::Number(NumberDataType::UInt64),
+                    Value::Column(NumberType::from_data(file_lens)),
+                ),
+                BlockEntry::new(
+                    DataType::Timestamp.wrap_nullable(),
+                    Value::Column(TimestampType::from_opt_data(file_last_modifieds)),
+                ),
+            ],
+            row_number,
+        )
+    }
+}
+
+const MAX_BATCH_SIZE: usize = 1000;
+
+#[async_trait::async_trait]
+impl AsyncSource for TempFilesTableAsyncSource {
```

Comment:
Maybe we could abstract a basic SourceClass named `FileListSource`, `copy into` or other features may need it.

Reply:
Alright, I'll try to leave some reusable code as a basis for potential refactoring in the future